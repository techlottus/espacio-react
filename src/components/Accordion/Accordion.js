import React, { useEffect } from "react";

export const Accordion = ({ data }) => {
  const accordionRef = React.createRef();

  useEffect(() => {
    accordionRef.current.data = {
      items: data.items || [],
    };
  });
  return (
    <>
      <lottus-accordion ref={accordionRef}></lottus-accordion>
    </>
  );
};
