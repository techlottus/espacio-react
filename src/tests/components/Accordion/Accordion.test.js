import React from "react";
import { mount } from "enzyme";
import { Accordion } from "../../../components/Accordion/Accordion";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState = {
  data: {
    items: null,
  },
};

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <Accordion data={initState.data} />
    </Provider>
  );
};

describe("Test <Accordion />", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });
});
