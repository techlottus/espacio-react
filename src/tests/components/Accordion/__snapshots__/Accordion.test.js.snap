// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Accordion /> data returns null 1`] = `<lottus-accordion />`;

exports[`Test <Accordion /> should init component 1`] = `<lottus-accordion />`;
